boards/FSM/Src/App/App_FlowMeter.c

```diff
@@ -13,17 +13,21 @@ struct FlowMeter *App_FlowMeter_Create(float (*get_flow_rate)(void))
 {
     assert(get_flow_rate != NULL);
 
-    static struct FlowMeter flow_meters[MAX_NUM_OF_FLOWMETERS];
-    static size_t           alloc_index = 0;
-
-    assert(alloc_index < MAX_NUM_OF_FLOWMETERS);
+    struct FlowMeter *flow_meter =
+        (struct FlowMeter *)malloc(sizeof(struct FlowMeter));
+    assert(flow_meter != NULL);
 
-    struct FlowMeter *const flow_meter = &flow_meters[alloc_index++];
-    flow_meter->get_flow_rate          = get_flow_rate;
+    flow_meter->get_flow_rate = get_flow_rate;
+    flow_meter->flow_rate     = NAN;
```

Comment:
@thekenu At start up do you think its a better idea to set the frequency as 0.0f of NaN?

Reply:
NaN